src/geofabrics/lidar_fetch.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Jul  2 10:10:55 2021
+
+@author: pearsonra
+"""
+
+import urllib
+import pathlib
+import requests
+import boto3
+import botocore
+import typing
+from . import geometry
+
+class OpenTopography:
+    """ A class to manage fetching LiDAR data from Open Topography
+    """
+    
+    SCHEME = "https"
+    NETLOC_API = "portal.opentopography.org"
+    PATH_API = "/API/otCatalog"
+    CRS = "EPSG:4326"
+    NETLOC_DATA = "opentopography.s3.sdsc.edu"
+    OT_BUCKET = 'pc-bulk'
+    
+    PATH_DATA = "/minio/pc-bulk/"
+    USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36'
+    
+    
+    
+    def __init__(self, catchment_geometry: geometry.CatchmentGeometry, cache_path: typing.Union[str, pathlib.Path], 
+                 redownload_files: bool = False, download_limit_gbytes: typing.Union[int, float] = 100, verbose: bool = False):
+        """ Load in lidar with relevant processing chain.
+        
+        Note in case of multiple datasets could select by name, 
+        spatial extent, or most recent. download_size is in GB. """
+        
+        self.catchment_geometry = catchment_geometry
+        self.cache_path = pathlib.Path(cache_path)
+        self.redownload_files = redownload_files
+        self.download_limit_gbytes = download_limit_gbytes
+        self.verbose = verbose
+        
+        self._dataset_prefixes = None
+        
+    def run(self):
+        """ Querey for  LiDAR data within a catchment and download any that hasn't already been downloaded """
+        self._dataset_prefixes = []
+        
+        json_response = self.query_for_datasets_inside_catchment()
+        
+        ot_endpoint_url = urllib.parse.urlunparse((self.SCHEME, self.NETLOC_DATA, "", "", "", ""))
+        client = boto3.client('s3', endpoint_url=ot_endpoint_url, 
+                              config=botocore.config.Config(signature_version=botocore.UNSIGNED))
+        
+        # cycle through each dataset within a region
+        for i in range(len(json_response['Datasets'])):
+            dataset_prefix = json_response['Datasets'][i]['Dataset']['alternateName']
```

Comment:
```suggestion
        for dataset_node in json_response['Datasets'])):
            dataset_prefix = dataset_node['Dataset']['alternateName']
```

changing for loop to for-each loop to improve readability. Find a variable name for `dataset_node` that suits you